src/main/java/com/pubnub/api/services/ExtendedPresenceService.java

```diff
@@ -33,7 +33,7 @@ Call<Envelope<JsonElement>> getState(@Path("subKey") String subKey,
                                          @QueryMap Map<String, String> options);
 
     @GET("v2/presence/sub-key/{subKey}/channel/{channel}/uuid/{uuid}/data")
-    Call<Envelope<JsonElement>> setState(@Path("subKey") String subKey,
+    Call<Envelope> setState(@Path("subKey") String subKey,
```

Comment:
Does it have to change?

Reply:
No. I modified 

  @GET("v2/presence/sub-key/{subKey}/channel/{channel}/heartbeat")
    Call<Envelope> heartbeat(@Path("subKey") String subKey,

  @GET("v2/presence/sub-key/{subKey}/channel/{channel}/heartbeat")
    Call<Envelope<**JsonElement**>> heartbeat(@Path("subKey") String subKey,

So now both setState(withHeartbeat=false) and setState(withHeartbeat=true) returns Envelopw<JsonElement>